src/uprobe/mod.rs

```diff
@@ -79,17 +88,13 @@ impl Uprobe {
                 message: "binary path is invalid".to_string(),
             });
         }
-        if self.symbol.is_none() {
-            return Err(BccError::InvalidUprobe {
-                message: "symbol is required".to_string(),
-            });
-        }
         let binary = binary.unwrap();
-        let symbol = self.symbol.unwrap();
+        let symbol = self.symbol.as_ref().map(|s| s.as_str()).unwrap_or("");
         let pid = self.pid.unwrap_or(-1);
         let handler = self.handler.unwrap();
+        let addr = self.addr;
 
-        let (path, addr) = crate::symbol::resolve_symbol_path(&binary, &symbol, 0x0, pid)?;
+        let (path, addr) = crate::symbol::resolve_symbol_path(&binary, &symbol, addr.unwrap_or(0x0), pid)?;
```

Comment:
Should we make changes to how this function works? Maybe it would make sense for it to take option types instead of having each call site handle unwrapping/defaults?